x-pack/plugins/triggers_actions_ui/public/application/sections/alerts_table/alerts_table.tsx

```diff
@@ -377,6 +379,68 @@ const AlertsTable: React.FunctionComponent<AlertsTableProps> = (props: AlertsTab
     return props.columns;
   }, [getCellActions, disabledCellActions, props.columns, visibleCellActions]);
 
+  // Merges the default grid style with the grid style that comes in through props.
+  // In addition, merge rowclasses and styles to allow for active flyout row to be highlighted.
+  // When a rowClassMapper is passed in, generate those classes and merge them with the active ones.
+  const actualGridStyle = useMemo(() => {
+    const propGridStyle: NonNullable<EuiDataGridStyle> = props.gridStyle ?? {};
+    // Merges default row classes, custom ones and adds the active row class style
+    const mergedGridStyle: EuiDataGridStyle = {
+      ...DefaultGridStyle,
+      ...propGridStyle,
+      rowClasses: activeRowClasses,
+    };
+
+    // Generate rowClasses from `rowClassMapper` when one was provided
+    let mappedRowClasses: EuiDataGridStyle['rowClasses'];
+    const rowClassMapper = props.rowClassMapper;
+    if (rowClassMapper) {
+      mappedRowClasses = alerts.reduce<NonNullable<EuiDataGridStyle['rowClasses']>>(
+        (rowClasses, alert, index) => {
+          const rowClass = rowClassMapper(alert);
+          if (rowClass) {
+            rowClasses[index + pagination.pageIndex * pagination.pageSize] = rowClass;
+          }
+          return rowClasses;
+        },
+        {}
+      );
+    }
+
+    // If ANY additional rowClasses have been provided, we need to merge them
+    if (propGridStyle.rowClasses || mappedRowClasses) {
+      // Get all row indices with a rowClass.
+      const mergedKeys = [
+        ...Object.keys(activeRowClasses),
+        ...Object.keys(propGridStyle.rowClasses || {}),
+        ...Object.keys(mappedRowClasses || {}),
+      ];
+      // Deduplicate keys to avoid extra iterations
+      const dedupedKeys = Array.from(new Set(mergedKeys));
+
+      // For each index, merge row classes
+      const mergedRowClasses = dedupedKeys.reduce<NonNullable<EuiDataGridStyle['rowClasses']>>(
+        (rowClasses, key) => {
+          const intKey = parseInt(key, 10);
+
+          // Concatenate class names for active row class and passed-in row class
+          // (if either exists)
+          rowClasses[intKey] = [
+            activeRowClasses[intKey],
+            propGridStyle.rowClasses?.[intKey],
+            mappedRowClasses?.[intKey],
+          ]
+            .filter(Boolean)
+            .join(' ');
+          return rowClasses;
+        },
+        {}
+      );
+      mergedGridStyle.rowClasses = mergedRowClasses;
+    }
+    return mergedGridStyle;
+  }, [alerts, activeRowClasses, props.rowClassMapper, pagination, props.gridStyle]);
+
```

Comment:
it would probably be easier to test then

Reply:
Agree, will move this piece of code to its own file. 

For testing however I would like to test the public interface instead of the internals. That would allow us to make sure the integration with the Data Grid doesn't break in the future.

I am talking to ResponseObs to see why the tests are not rendering a proper Data Grid. Once that's sorted, we should be able to write proper tests.

In case the test renderer is not easy to fix, I can still come back and add unit tests for only this part of the logic. Wdyt?

Reply:
With the latest "refactor" this has split up a bit. Do you still think it makes sense to movie it outside now?